src/amber/cli/commands/pipelines.cr

```diff
@@ -0,0 +1,132 @@
+require "cli"
+require "shell-table"
+require "../helpers/sentry"
+
+module Amber::CLI
+  class MainCommand < ::Cli::Supercommand
+    class Pipelines < Command
+      getter result = Array(NamedTuple(pipes: Array(String), plugs: Array(String))).new
+      property current_pipe : String?
+
+      class Options
+        bool "--no-color", desc: "Disable colored output", default: false
+        bool "--no-plugs", desc: "Don't output the plugs", default: false
+        help
+      end
+
+      class BadRoutesException < Exception
+      end
+
+      ROUTES_PATH          = "config/routes.cr"
+      LABELS               = %w(Pipe Plug)
+      LABELS_WITHOUT_PLUGS = %w(Pipe)
+
+      PIPELINE_REGEX =
+        /^
+          \s*
+          pipeline  # match 'pipeline'
+          \s+       # require at least one whitespace character after 'pipeline'
+          (
+            (?:
+              (?:
+                \:(?:\w+)
+                |
+                \"(?:\w+)\"
+              )
+              (?:\,\s*)?
+            )+
+          )         # match and capture all contiguous words
+        /x
+
+      PLUG_REGEX =
+        /^
+          \s*
+          plug        # match 'plug'
+          \s+         # require at least one whitespace character after 'plug'
+          (
+            [\w:]+    # match at least one words with maybe a colon
+          )?
+          (?:
+            [\.\s*\(] # until we reach '.', spaces, or braces
+          )?
+        /x
+
+      FAILED_TO_PARSE_ERROR = "Could not parse pipeline/plugs in #{ROUTES_PATH}"
```

Comment:
@NeverHappened Looks like syntax highlighting is broken here, I tried to remove quotes (`'`) in comments and syntax highlighting worked again:

With quotes (`'`) in comments

![screenshot_20180131_112412](https://user-images.githubusercontent.com/3067335/35634270-441d4a2c-0679-11e8-8df2-6fcd121089b3.png)

Without quotes (`'`) in comments:

![screenshot_20180131_112500](https://user-images.githubusercontent.com/3067335/35634308-60c71a18-0679-11e8-97f9-557e06de587b.png)